tests/ragged_tests.py

```diff
@@ -714,13 +714,27 @@ def test_subset_callable(self):
         self.assertTrue(all(ds_sub["id"] == [1, 2]))
         self.assertTrue(all(ds_sub["rowsize"] == [5, 4]))
 
+        func = lambda arr1, arr2: np.logical_and(
```

Comment:
Could you create another test function with the following.

Reply:
You mean having the test case as a separate method? Sure!

Reply:
Yes, to test one specific aspect per method.

Reply:
We should  consider testing the following scenarios:

* a dimension is passed in the tuple that doesn't exist (separate test) in the dataset. I believe if we try to do this we will probably get a KeyError or xarrays Dataset equivalent.
* a user passes in a callable that takes in less parameters than the size of the tuple. When unpacking the iterable it will pass more arguments than the function definition expects.
* type errors in the case where the function parameters expected type is different from its actual type.